### DISCUSSION TITLE
[Adonis 5] Route Validator

### DISCUSSION THREAD

**Discussion by Melchyore** [adonisjs/core#1435] [Help] [answered]
Hi,

In the previous version (i.e 4.1), we were able to use Route Validator, but now, it seems that the validator method does not exist anymore in the Route class, is it intended? If yes, then can we create a validator class? If yes, how can we achieve that?

**Comment by fnoquiq** [comment, marked as answer]
Hey, validations can be done on controllers through the request context

```ts
import StoreValidator from 'App/Validators/userStore'

export default class UsersController {
  public async store({ request }: HttpContextContract) {
    const { username, password } = await request.validate(StoreValidator)

    const user = await User.create({
      username,
      password,
    })

    return user
  }
}
```

You can find the documentation [here](https://preview.adonisjs.com/guides/http/form-submissions#using-validator-classes)

**Reply by Melchyore** [reply]
@fnoquiq thanks, it works like a charm! I was reading the validator guide...
